fe/fe-core/src/main/java/com/starrocks/sql/optimizer/CachingMvPlanContextBuilder.java

```diff
@@ -23,14 +23,15 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 
 public class CachingMvPlanContextBuilder {
 
     private static final Logger LOG = LogManager.getLogger(CachingMvPlanContextBuilder.class);
     private static final CachingMvPlanContextBuilder INSTANCE = new CachingMvPlanContextBuilder();
 
-    private Cache<MaterializedView, MvPlanContext> mvPlanContextCache = buildCache();
+    private Cache<MaterializedView, List<MvPlanContext>> mvPlanContextCache = buildCache();
```

Comment:
How many plans would a mv have ?

Reply:
now it will change from one to two, maybe it will be more later. because Query may has more than one view, we can decide whether to return each  view as a result of plan, which will be 2^n - 1  cases(n is the number of views)

Reply:
2^N not realistic, let's keep it 2